source/fe/mapping_q_generic.cc

```diff
@@ -3877,8 +3833,16 @@ compute_mapping_support_points(const typename Triangulation<dim,spacedim>::cell_
         // manifold, otherwise compute them from the points around it
         if (dim != spacedim)
           add_quad_support_points(cell, a);
+        else if (dynamic_cast<const Boundary<dim,spacedim> *>(&cell->get_manifold()) == NULL)
+          {
+            std::vector<Point<spacedim> > quad_points (Utilities::fixed_power<dim>(polynomial_degree-1));
+            get_intermediate_points_on_object(cell->get_manifold(), line_support_points, cell, quad_points);
+            for (unsigned int i=0; i<quad_points.size(); ++i)
+              a.push_back(quad_points[i]);
+          }
         else
-          add_weighted_interior_points (support_point_weights_on_quad, a);
+          add_weighted_interior_points(support_point_weights_on_quad, a);
```

Comment:
(I will fix the indentation once we are ready with discussing the code, otherwise my previous comment will get hidden again.)